src/components/QuranReader/TafsirView/TafsirBody.tsx

```diff
@@ -0,0 +1,211 @@
+  // we need to set this `initialTafsirId` as a selectedTafsirId
+  // we did not use `useState(initialTafsirId)` because `useRouter`'s query string is undefined on first render
+  useEffect(() => {
+    if (initialTafsirId) {
+      setSelectedTafsirId(initialTafsirId);
+    }
+  }, [initialTafsirId]);
+
+  const onTafsirSelected = useCallback(
+    (id: number) => {
+      setSelectedTafsirId(id);
+      fakeNavigate(
+        getVerseTafsirNavigationUrl(
+          Number(selectedChapterId),
+          Number(selectedVerseNumber),
+          id.toString(),
+        ),
+      );
+      dispatch(
+        setSelectedTafsirs({
+          tafsirs: [id],
+          locale: lang,
+        }),
+      );
+    },
+    [dispatch, lang, selectedChapterId, selectedVerseNumber],
+  );
+
+  const { data: tafsirListData } = useSWR<TafsirsResponse>(makeTafsirsUrl(lang), fetcher);
+
+  // selectedLanguage is based on selectedTafir's language
+  // but we need to fetch the data from the API first to know what is the lanaguage of `selectedTafsirId`
+  // so we get the data from the API and set the selectedLanguage once it is loaded
+  useEffect(() => {
+    if (tafsirListData) {
+      const languageName = getSelectedTafsirLanguage(tafsirListData, selectedTafsirId);
+      setSelectedLanguage(languageName);
+    }
+  }, [onTafsirSelected, selectedTafsirId, tafsirListData]);
+
+  // there's no 1:1 data that can map our locale options to the tafsir language options
+  // so we're using options that's available from tafsir for now
+  // TODO: update lanague options, to use the same options as our LanguageSelector
+  const languageOptions = tafsirListData ? getTafsirsLanguageOptions(tafsirListData.tafsirs) : [];
+
+  const renderTafsir = useCallback((data) => {
+    if (!data) return <TafsirSkeleton />;
+    const tafsirVerses = data?.tafsir.verses;
+    const htmlText = data?.tafsir.text;
+    const words = Object.values(tafsirVerses).map(getVerseWords).flat();
+
+    if (!htmlText) return <TafsirSkeleton />;
+
+    const [firstVerseKey, lastVerseKey] = getFirstAndLastFirstKeys(tafsirVerses);
+    return (
+      <div>
+        {Object.values(tafsirVerses).length > 1 && (
+          <TafsirGroupMessage from={firstVerseKey} to={lastVerseKey} />
+        )}
+        <div className={styles.verseTextContainer}>
+          <VerseText words={words} />
+        </div>
+        <div className={styles.separatorContainer}>
+          <Separator />
+        </div>
+        <div dangerouslySetInnerHTML={{ __html: htmlText }} />
+      </div>
+    );
+  }, []);
+
+  const selectedTafsirLanguage = getSelectedTafsirLanguage(tafsirListData, selectedTafsirId);
+
+  const tafsirContentQuerykey = makeTafsirContentUrl(selectedTafsirId, selectedVerseKey, {
+    words: true,
+    ...getDefaultWordFields(quranReaderStyles.quranFont),
+  });
+
+  return (
+    <div dir={isRTLLanguage(selectedTafsirLanguage) ? 'rtl' : 'ltr'}>
```

Comment:
rtl and ltr should only be set on the tafsir html content. When I mentioned "body" I haven't reviewed the code by that time, apologies